cfgov/v1/models/base.py

```diff
@@ -149,73 +149,74 @@ def generate_view_more_url(self, request):
         from v1.models.learn_page import AbstractFilterPage
         related = {}
-        query = models.Q(('tags__name__in', self.tags.names()))
         search_types = [
-            ('blog', 'posts', 'Blog', query),
-            ('newsroom', 'newsroom', 'Newsroom', query),
-            ('events', 'events', 'Events', query),
+            ('blog', block.value.get('relate_posts')),
+            ('newsroom', block.value.get('relate_newsroom')),
+            ('events', block.value.get('relate_events')),
         ]
-
-        def fetch_children_by_specific_category(block, parent_slug):
-            """
-            This used to be a Page.objects.get, which would throw
-            an exception if the requested parent wasn't found. As of
-            Django 1.6, you can now do Page.objects.filter().first();
-            the advantage here is that you can check for None right
-            away and not have to rely on catching exceptions, which
-            in any case didn't do anything useful other than to print
-            an error message. Instead, we just return an empty query
-            which has no effect on the final result.
-            """
-            parent = Page.objects.filter(slug=parent_slug).first()
-            if parent:
-                child_query = Page.objects.child_of_q(parent)
-                if 'specific_categories' in block.value:
-                    child_query &= specific_categories_query(
-                        block, parent_slug)
-            else:
-                child_query = Q()
-            return child_query
-
-        def specific_categories_query(block, parent_slug):
-            specific_categories = ref.related_posts_category_lookup(
-                block.value['specific_categories']
-            )
-            choices = [c[0] for c in ref.choices_for_page_type(parent_slug)]
-            categories = [c for c in specific_categories if c in choices]
-            if categories:
-                return Q(('categories__name__in', categories))
-            else:
-                return Q()
-
-        for parent_slug, search_type, search_type_name, search_query in \
-                search_types:
-            search_query &= fetch_children_by_specific_category(
-                block, parent_slug)
-            if parent_slug == 'events':
-                search_query |= fetch_children_by_specific_category(
-                    block, 'archive-past-events') & query
-            relate = block.value.get('relate_{}'.format(search_type), None)
-            if relate:
-                type_query = (
-                    AbstractFilterPage.objects.live().filter(
-                        search_query
-                    ).distinct().exclude(id=self.id).order_by(
-                        '-date_published'
-                    )
-                )
+        specific_categories = block.value['specific_categories']
+
+        for slug, should_get_related_posts in search_types:
+            if should_get_related_posts:
```

Comment:
Could this be a guard clause? So return immediately if `should_get_related_posts` does not exist. I don't know what that would look like in python.

Reply:
I changed this in another way, but one option could've been to do `if not should_get_related_posts: continue` and then have all the code below that indented to the left.